service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/ChildControllerMidEventTest.java

```diff
@@ -0,0 +1,219 @@
+    }
+
+    @Test
+    void shouldReturnNoErrorsWhenAllDatesOfBirthAreTodayOrInPast() throws Exception {
+        ZonedDateTime today = ZonedDateTime.now();
+        ZonedDateTime yesterday = today.minusDays(1);
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = makeRequest(
+            new OldChildren(createChild(today), createChild(yesterday))
+        );
+        assertThat(callbackResponse.getErrors()).doesNotContain(ERROR_MESSAGE);
+    }
+
+    @Test
+    void shouldReturnDateOfBirthErrorsForNewChildWhenFutureDateOfBirth() throws Exception {
+        CallbackRequest request = CallbackRequest.builder()
+            .caseDetails(CaseDetails.builder()
+                .id(12345L)
+                .data(ImmutableMap.of(
+                    "children1", ImmutableList.of(
+                        ImmutableMap.of(
+                            "id", "",
+                            "value", Child.builder()
+                                .party(ChildParty.builder()
+                                    .dateOfBirth(Date.from(ZonedDateTime.now().plusDays(1).toInstant()))
+                                    .build())
+                                .build()
+                        )
+                    )
+                ))
+                .build())
+            .build();
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = makeRequest(request);
+
+        assertThat(callbackResponse.getErrors()).contains(ERROR_MESSAGE);
+    }
+
+    @Test
+    void shouldReturnDateOfBirthErrorsForNewChildrenWhenThereIsMultipleChildren() throws Exception {
+        CallbackRequest request = CallbackRequest.builder()
+            .caseDetails(CaseDetails.builder()
+                .id(12345L)
+                .data(ImmutableMap.of(
+                    "children1", ImmutableList.of(
+                        ImmutableMap.of(
+                            "id", "",
+                            "value", Child.builder()
+                                .party(ChildParty.builder()
+                                    .dateOfBirth(Date.from(ZonedDateTime.now().plusDays(1).toInstant()))
+                                    .build())
+                                .build()
+                        ),
+                        ImmutableMap.of(
+                            "id", "",
+                            "value", Child.builder()
+                                .party(ChildParty.builder()
+                                    .dateOfBirth(Date.from(ZonedDateTime.now().plusDays(1).toInstant()))
+                                    .build())
+                                .build()
+                        )
+                    )
+                ))
+                .build())
+            .build();
+
+        AboutToStartOrSubmitCallbackResponse callbackResponse = makeRequest(request);
+
+        assertThat(callbackResponse.getErrors()).containsExactly(ERROR_MESSAGE);
```

Comment:
Any particular reason why we are using containsExactly here and in others contains?

Reply:
I prefer containsExactly, a little stricter

Reply:
have used containsOnlyOnce 😎